src/pipelines/master_pipeline.py

```diff
@@ -16,14 +19,17 @@ def main(argv):
     controls the over-arching implmentation of the algorithms
     '''
     directory, features, algorithms, parameters = argv
+    
+    # Create a numpy random state
+    random_state = np.random.RandomState(parameters['seed'])
 
     #parsing
     print("parsing json data...",file=sys.stderr)
-    clusters, order, data, test_clusters, test_order, test_data, corpusdict = parse_json.main([directory, parameters])
+    clusters, order, data, test_clusters, test_order, test_data, corpusdict = parse_json.main(directory, parameters)
 
     #preprocessing
     print("preprocessing...",file=sys.stderr)
-    vocab, encoder_decoder, lda_model, tf_model, w2v_model = preprocess.main([features, parameters, corpusdict, data])
+    vocab, encoder_decoder, lda_model, tf_model, w2v_model = preprocess.main(features, parameters, corpusdict, data)
```

Comment:
Hm yes, another issue is that if any other optional code paths in preprocess.main were to use random_state, then pseudo-random behavior would be dependent on what combinations of parameters were requested, in what order... along the lines of what you're suggesting, passing in the seed to build_lda would probably fix that.